From benh@kernel.crashing.org Wed Dec 27 20:27:34 2006
From: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date: Thu, 28 Dec 2006 15:27:27 +1100
Subject: USB: Fix EHCI warning
To: akpm@osdl.org
Cc: Greg KH <greg@kroah.com>
Message-ID: <1167280047.23340.58.camel@localhost.localdomain>


This patch fixes a warning introduced by the big endian MMIO EHCI
support patch on platforms that don't have readl_be/writel_be variants
(though mostly harmless as those are called in an if (0) statement,
but gcc still warns).

Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

---
 drivers/usb/host/ehci.h |    8 ++++++++
 1 file changed, 8 insertions(+)

--- gregkh-2.6.orig/drivers/usb/host/ehci.h
+++ gregkh-2.6/drivers/usb/host/ehci.h
@@ -671,17 +671,25 @@ ehci_port_speed(struct ehci_hcd *ehci, u
 static inline unsigned int ehci_readl (const struct ehci_hcd *ehci,
 				       __u32 __iomem * regs)
 {
+#ifdef CONFIG_USB_EHCI_BIG_ENDIAN_MMIO
 	return ehci_big_endian_mmio(ehci) ?
 		readl_be((__force u32 *)regs) :
 		readl((__force u32 *)regs);
+#else
+	return readl((__force u32 *)regs);
+#endif
 }
 
 static inline void ehci_writel (const struct ehci_hcd *ehci,
 				const unsigned int val, __u32 __iomem *regs)
 {
+#ifdef CONFIG_USB_EHCI_BIG_ENDIAN_MMIO
 	ehci_big_endian_mmio(ehci) ?
 		writel_be(val, (__force u32 *)regs) :
 		writel(val, (__force u32 *)regs);
+#else
+	writel(val, (__force u32 *)regs);
+#endif
 }
 
 /*-------------------------------------------------------------------------*/
